src/app/components/Byline/utilities/bylineExtractor/index.tsx

```diff
@@ -26,47 +26,52 @@ const pathOrZeroIndexModelBlocks = (
 
-  const authorBlock = bylineBlocks.find(block => block.type === 'name');
-  const jobRoleBlock = bylineBlocks.find(block => block.type === 'role');
-  const twitterBlock = bylineBlocks.find(block => block.type === 'link');
-  const locationBlock = bylineBlocks.find(block => block.type === 'location');
-  const imagesBlock = bylineBlocks.find(block => block.type === 'images');
+    const authorBlock = bylineBlocks.find(block => block.type === 'name');
+    const jobRoleBlock = bylineBlocks.find(block => block.type === 'role');
+    const twitterBlock = bylineBlocks.find(block => block.type === 'link');
+    const locationBlock = bylineBlocks.find(block => block.type === 'location');
+    const imagesBlock = bylineBlocks.find(block => block.type === 'images');
 
-  const authorName = pathOrZeroIndexModelBlocks(2, 'text', authorBlock);
-  const jobRole = pathOrZeroIndexModelBlocks(2, 'text', jobRoleBlock);
+    const authorName = pathOrZeroIndexModelBlocks(2, 'text', authorBlock);
+    const jobRole = pathOrZeroIndexModelBlocks(2, 'text', jobRoleBlock);
+    if (!authorName) {
+      return null;
+    }
 
-  if (!authorName) {
-    return null;
-  }
+    const twitterText = pathOrZeroIndexModelBlocks(2, 'text', twitterBlock);
+    const twitterLink = pathOrZeroIndexModelBlocks(3, 'locator', twitterBlock);
+    const location = pathOrZeroIndexModelBlocks(2, 'text', locationBlock);
+    const locator = pathOrZeroIndexModelBlocks(2, 'locator', imagesBlock);
+    const originCode = pathOrZeroIndexModelBlocks(2, 'originCode', imagesBlock);
 
-  const twitterText = pathOrZeroIndexModelBlocks(2, 'text', twitterBlock);
-  const twitterLink = pathOrZeroIndexModelBlocks(3, 'locator', twitterBlock);
-  const location = pathOrZeroIndexModelBlocks(2, 'text', locationBlock);
-  const locator = pathOrZeroIndexModelBlocks(2, 'locator', imagesBlock);
-  const originCode = pathOrZeroIndexModelBlocks(2, 'originCode', imagesBlock);
+    let authorImage = buildIChefURL({
+      originCode,
+      locator,
+      resolution: 160,
+    });
 
-  let authorImage = buildIChefURL({
-    originCode,
-    locator,
-    resolution: 160,
-  });
+    if (!authorImage.endsWith('.png.webp')) authorImage = '';
 
-  if (!authorImage.endsWith('.png.webp')) authorImage = '';
+    const contributorBlock = blocks?.[0] ?? [];
+    const authorTopicUrl = contributorBlock?.model?.topicUrl ?? '';
 
-  const contributorBlock = blocks?.[0] ?? [];
-  const authorTopicUrl = contributorBlock?.model?.topicUrl ?? '';
+    return {
+      authorName,
+      jobRole,
+      twitterText,
+      twitterLink,
+      authorImage,
+      location,
+      authorTopicUrl,
+    };
+  });
 
-  return {
-    authorName,
-    jobRole,
-    twitterText,
-    twitterLink,
-    authorImage,
-    location,
-    authorTopicUrl,
-  };
+  return bylineValues.includes(null) || bylineValues.length === 0
+    ? null
+    : bylineValues.filter(item => item !== null);
```

Comment:
I think there might be an even cleaner way of doing this. `filter(Boolean)` on an array will remove any null entries. It will mean however, that the result will always be an array. But hopefully we can cater for that in the code which calls this?

```
return bylineValues.map({... mapping code here ...}).filter(Boolean);
```

Reply:
Yeah this is cleaner ... can update the places that are checking for a null value to check for an empty array instead i suppose!

Reply:
the issue with this way is typescript doesn't seem to recognise that this array will now definitely not have null values within it, so the type has to be updated to (BylineLinkedData | null)[]; , which is not technically true!

Reply:
OK - happy to update the type if it makes this code a bit simpler.